ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/ValueValidator.scala

```diff
@@ -100,28 +91,42 @@ object ValueValidator {
     case Sum.Optional(o) =>
       o.value.fold[Either[StatusRuntimeException, domain.Value]](Right(Lf.ValueNone))(
         validateValue(_).map(v => Lf.ValueOptional(Some(v))))
-    case Sum.Map(m) =>
-      val entries = m.entries
+    case Sum.Map(map0) =>
+      val map = map0.entries
         .foldLeft[Either[StatusRuntimeException, FrontStack[(String, domain.Value)]]](
           Right(FrontStack.empty)) {
-          case (acc, ApiMap.Entry(key, value0)) =>
+          case (acc, api.Map.Entry(key, value0)) =>
             for {
               tail <- acc
               v <- requirePresence(value0, "value")
               validatedValue <- validateValue(v)
             } yield (key -> validatedValue) +: tail
         }
-
       for {
-        list <- entries
-        map <- SortedLookupList.fromImmArray(list.toImmArray).left.map(invalidArgument)
+        entries <- map
+        map <- SortedLookupList.fromImmArray(entries.toImmArray).left.map(invalidArgument)
       } yield Lf.ValueMap(map)
 
+    case Sum.GenMap(genMap0) =>
+      val genMap = genMap0.entries
+        .foldLeft[Either[StatusRuntimeException, BackStack[(domain.Value, domain.Value)]]](
+          Right(BackStack.empty)) {
+          case (acc, api.GenMap.Entry(key0, value0)) =>
+            for {
+              stack <- acc
+              key <- requirePresence(key0, "key")
+              validatedKey <- validateValue(key)
+              value <- requirePresence(value0, "value")
```

Comment:
We can potentially avoid some work by checking for the presence of the `value` field before validating the `key` value.

Reply:
I inversed line 118 and 119.